### PR TITLE
Horizontal bar chart with axis - component Tests

### PR DESCRIPTION
**Pull request by v-baambati**
Added component tests for HorizontalBarChartWithAxis increasing code coverage from 64% to 78%

Before:
HBCWA-before

After:
HBCWAAfter

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/0bf811dbcaaa6b78743537d142fe4d348b457045'>0bf811dbcaaa6b78743537d142fe4d348b457045</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/react-charting/docs/TestPlans/HorizontalBarChartWithAxis/ComponentTests.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+1. **Bar: Bar data, Bar color, bar label**
+1. **Callout: Default/custom callout**
+1. **Labels: Default/Custom labels**
+1. **Legend: show/hide legends, highlight the corresponding bars on legend hover**
+
+|                                    **Test steps**                                     |                          **Validation**                           | **Tool used** |
+| :-----------------------------------------------------------------------------------: | :---------------------------------------------------------------: | :-----------: |
+|                              Test 1: [Snapshot testing]                               |                                                                   |               |
+|                     - With only data prop, string data on y-axis.                     |      Should render horizontal bar chart with axis correctly       |      RTL      |
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/HorizontalBarChartWithAxis/ComponentTests.md:13]: this chart type supports only numeric data on y axis

File: packages/react-charting/docs/TestPlans/HorizontalBarChartWithAxis/ComponentTests.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+|                  - Specify custom callout and hover mouse over a bar                  |           Should show the custom callout over that bar            |      RTL      |
+|                   - Specify numeric axis data to display in callout                   |           Should show the callout with axis data on it            |      RTL      |
+|      - Specify string yaxis data and numeric xaxis data to display it in callout      |          Should show the callout data based on axis data          |      RTL      |
+|                            Test 8: [Sub-Component]: Labels                            |                                                                   |               |
+|          - Set showYAxisLabelTooTip to “true” and mouse hover on Yaxis label          |         Should show y axis tooltip when mouse hover on it         |      RTL      |
+|                            - Set showYAxisLabels to “true”                            |         Should show y axis labels data without truncating         |      RTL      |
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/HorizontalBarChartWithAxis/ComponentTests.md:47]: Add a test setting expandYAxisLabels to true

Comment (AtishayMsft) [reply packages/react-charting/docs/TestPlans/HorizontalBarChartWithAxis/ComponentTests.md:47]: It is called showYAxisLables

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**